src/components/image/image.tsx

```diff
@@ -0,0 +1,103 @@
+import { Component, h, Prop } from '@stencil/core';
+import { Image } from '../../global/shared-types/image.types';
+
+/**
+ * Image is a component that displays an image.
+ * It supports lazy loading of images to improve performance,
+ * the image will only be loaded when it enters the viewport.
+ *
+ * @private
+ *
+ * @exampleComponent limel-example-image
+ * @exampleComponent limel-example-image-lazy
+ * @exampleComponent limel-example-image-interactive
+ * @exampleComponent limel-example-image-gallery
+ * @exampleComponent limel-example-image-no-source
+ */
+@Component({
+    tag: 'limel-image',
+    shadow: true,
+    styleUrl: 'image.scss',
+})
+export class ImageComponent {
+    /**
+     * The image object containing src, alt, and loading properties
+     */
+    @Prop()
+    public image: Image;
+
+    private intersectionObserver: IntersectionObserver | null = null;
+    private currentImageElement: HTMLImageElement | null = null;
+
+    public componentDidRender() {
+        this.setupImageIntersectionObserver();
+    }
```

Comment:
If we set this up every time we rerender (instead of componentDidLoad), do we need to disconnect the old observer first?